diff --git a/src/database/manager/parser/liberty/CMakeLists.txt b/src/database/manager/parser/liberty/CMakeLists.txt
index f6511c3..43f18b5 100644
--- a/src/database/manager/parser/liberty/CMakeLists.txt
+++ b/src/database/manager/parser/liberty/CMakeLists.txt
@@ -19,9 +19,7 @@ endif()
 
 message(STATUS "liberty parser rust lib path ${RUST_LIB_PATH}")
 
-ADD_EXTERNAL_PROJ(liberty)
-
-target_link_libraries(liberty str sta-solver log ${RUST_LIB_PATH} dl)
+target_link_libraries(liberty str sta-solver log liberty_parser dl)
 
 target_include_directories(liberty PUBLIC 
     ${CMAKE_CURRENT_SOURCE_DIR} 
diff --git a/src/database/manager/parser/spef/CMakeLists.txt b/src/database/manager/parser/spef/CMakeLists.txt
index 322eb15..3ab9801 100644
--- a/src/database/manager/parser/spef/CMakeLists.txt
+++ b/src/database/manager/parser/spef/CMakeLists.txt
@@ -19,9 +19,7 @@ endif()
 
 message(STATUS "spef parser rust lib path ${RUST_LIB_PATH}")
 
-ADD_EXTERNAL_PROJ(spef)
-
-target_link_libraries(spef log ${RUST_LIB_PATH} dl)
+target_link_libraries(spef log spef_parser dl)
 target_include_directories(spef PUBLIC 
     ${CMAKE_CURRENT_SOURCE_DIR})
 
diff --git a/src/database/manager/parser/vcd/CMakeLists.txt b/src/database/manager/parser/vcd/CMakeLists.txt
index 0aa2635..ab944c2 100644
--- a/src/database/manager/parser/vcd/CMakeLists.txt
+++ b/src/database/manager/parser/vcd/CMakeLists.txt
@@ -15,9 +15,7 @@ SETUP_RUST_PROJECT()
 
 message(STATUS "vcd parser rust lib path ${RUST_LIB_PATH}")
 
-ADD_EXTERNAL_PROJ(vcd)
-
-target_link_libraries(vcd ${RUST_LIB_PATH} dl)
+target_link_libraries(vcd vcd_parser dl)
 target_include_directories(vcd PUBLIC 
     ${CMAKE_CURRENT_SOURCE_DIR})
 
diff --git a/src/database/manager/parser/verilog/CMakeLists.txt b/src/database/manager/parser/verilog/CMakeLists.txt
index 1ceaee3..a318da4 100644
--- a/src/database/manager/parser/verilog/CMakeLists.txt
+++ b/src/database/manager/parser/verilog/CMakeLists.txt
@@ -40,9 +40,7 @@ endif()
 
 message(STATUS "verilog parser rust lib path ${RUST_LIB_PATH}")
 
-ADD_EXTERNAL_PROJ(verilog)
-
-target_link_libraries(verilog str log ${ZLIB_LIBRARIES} ${RUST_LIB_PATH} dl) 
+target_link_libraries(verilog str log ${ZLIB_LIBRARIES} verilog_parser dl)
 
 target_include_directories(verilog
     PUBLIC
